templates/shared_services/airlock-notifier/porter.yaml

```diff
@@ -0,0 +1,158 @@
+registry: azuretre
+dockerfile: Dockerfile.tmpl
+
+credentials:
+  - name: azure_tenant_id
+    env: ARM_TENANT_ID
+  - name: azure_subscription_id
+    env: ARM_SUBSCRIPTION_ID
+  - name: azure_client_id
+    env: ARM_CLIENT_ID
+  - name: azure_client_secret
+    env: ARM_CLIENT_SECRET
+
+parameters:
+  - name: tre_id
+    type: string
+    description: "The ID of the parent TRE instance e.g., mytre-dev-3142"
+  - name: id
+    type: string
+    description: "Resource ID"
+  - name: tfstate_resource_group_name
+    type: string
+    description: "Resource group containing the Terraform state storage account"
+  - name: tfstate_storage_account_name
+    type: string
+    description: "The name of the Terraform state storage account"
+  - name: tfstate_container_name
+    type: string
+    default: "tfstate"
+    description: "The name of the Terraform state storage container"
+  - name: smtp_server_address
+    env: SMTP_SERVER_ADDRESS
+    type: string
+  - name: smtp_username
+    env: SMTP_USERNAME
+    type: string
+  - name: smtp_password
+    env: SMTP_PASSWORD
+    type: string
+  - name: smtp_from_email
+    env: SMTP_FROM_EMAIL
+    type: string
+  - name: arm_use_msi
+    env: ARM_USE_MSI
+    type: boolean
+    default: false
+
+mixins:
+  - exec
+  - az
+  - terraform:
+      clientVersion: 1.1.5
+      initFile: providers.tf
+
+install:
+  - terraform:
+      description: "Deploy shared service"
+      input: false
+      vars:
+        tre_id: "{{ bundle.parameters.tre_id }}"
+        tre_resource_id: "{{ bundle.parameters.id }}"
+        smtp_server_address: "{{ bundle.parameters.smtp_server_address }}"
+        smtp_username: "{{ bundle.parameters.smtp_username }}"
+        smtp_password: "{{ bundle.parameters.smtp_password }}"
+        smtp_from_email: "{{ bundle.parameters.smtp_from_email }}"
+        arm_use_msi: "{{ bundle.parameters.arm_use_msi }}"
+      backendConfig:
+        resource_group_name: "{{ bundle.parameters.tfstate_resource_group_name }}"
+        storage_account_name: "{{ bundle.parameters.tfstate_storage_account_name }}"
+        container_name: "{{ bundle.parameters.tfstate_container_name }}"
+        key: "{{ bundle.parameters.tre_id }}-shared-airlock-notifier"
+      outputs:
+        - name: airlock_notifier_logic_app_name
+        - name: airlock_notifier_logic_app_resource_group_name
+
+  - az:
+      description: "Login to Azure"
+      arguments:
+        - login
+      flags:
+        identity:
+        username: "{{ bundle.credentials.azure_client_id }}"
+
+  - az:
+      description: "Deploy logic app"
+      arguments:
+        - functionapp
+        - deployment
+        - source
+        - config-zip
+      flags:
+        name: "{{ bundle.outputs.airlock_notifier_logic_app_name }}"
+        resource-group: "{{ bundle.outputs.airlock_notifier_logic_app_resource_group_name }}"
+        subscription: "{{ bundle.credentials.azure_subscription_id }}"
+        src: LogicApp.zip
+
+
+upgrade:
```

Comment:
Do we support upgrades?

Reply:
Json changes in the logic app workflow definiton counts as an upgrade?

Reply:
Dropping upgrade support since we don't save SMTP properties